File: benchmark/c/cpw/find_first_natural_number_whose_factorial_divisible_x/find_first_natural_number_whose_factorial_divisible_x.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int find_first_natural_number_whose_factorial_divisible_x ( int x ) ;
int find_first_natural_number_whose_factorial_divisible_x ( int x ) {
  int i = 1;
  int fact = 1;
  for ( i = 1;
  i < x;
  i ++ ) {
    fact = fact * i;
    if ( fact % x == 0 ) break;
  }
  return i;
}

File: translation_result/else_bms/success/c/cpw/find_first_natural_number_whose_factorial_divisible_x/gpt4/CAPR/find_first_natural_number_whose_factorial_divisible_x.rs

fn find_first_natural_number_whose_factorial_divisible_x(x: i32) -> i32 {
    if x <= 0 {
        return 1;
    }
    let mut fact: i32 = 1;
    for j in 1..=x {
        fact = fact.wrapping_mul(j);
        if fact % x == 0 {
            return j;
        }
    }
    x
}
